components/tabs/tab-autocollapse-titles.js

```diff
@@ -0,0 +1,511 @@
+  [items, selected]
+  );
+
+  const data = React.useMemo(() =>
+    getTabTitles({
+      items,
+      selected,
+      onSelect,
+      collapsed: true
+    }).map(tab => ({
+      template: tab,
+      key: tab.key,
+      rgItemType: ListProps.Type.CUSTOM,
+      className: styles.popupMenuItem,
+      disabled: tab.props.disabled
+    })), [items, onSelect, selected]);
+
+  const popupAnchor = React.useMemo(() => (
+    <AnchorLink
+      moreClassName={moreClassName}
+      moreActiveClassName={moreActiveClassName}
+      hasActiveChildren={hasActiveChild}
+    />
+  ), [hasActiveChild, moreActiveClassName, moreClassName]);
+
+  const popup = React.useMemo(() => (
+    <PopupMenu
+      directions={morePopupDirections}
+      className={morePopupClassName}
+      data={data}
+      onSelect={onSelectHandler}
+    />
+  ), [data, morePopupClassName, onSelectHandler]);
+
+  if (items.length === 0) {
+    return null;
+  }
+
+  return (
+    <div className={classNames(styles.moreButton, styles.title)}>
+      <Dropdown
+        hoverMode
+        anchor={popupAnchor}
+      >
+        {popup}
+      </Dropdown>
+    </div>
+  );
+});
+
+MoreButton.propTypes = {
+  children: PropTypes.node,
+  items: PropTypes.array,
+  selected: PropTypes.string,
+  onSelect: PropTypes.func,
+  toMeasure: PropTypes.bool,
+  moreClassName: PropTypes.string,
+  moreActiveClassName: PropTypes.string,
+  morePopupClassName: PropTypes.string
+};
+
+MoreButton.displayName = 'MoreButton';
+
+const FakeMoreButton = React.memo(({
+  moreClassName,
+  moreActiveClassName,
+  hasActiveChildren
+}) => (
+  <div className={classNames(styles.moreButton, styles.title)}>
+    <AnchorLink
+      moreClassName={moreClassName}
+      moreActiveClassName={moreActiveClassName}
+      hasActiveChildren={hasActiveChildren}
+      tabIndex={-1}
+      disabled
+    />
+  </div>
+));
+FakeMoreButton.propTypes = {
+  moreClassName: PropTypes.string,
+  moreActiveClassName: PropTypes.string,
+  hasActiveChildren: PropTypes.bool
+};
+FakeMoreButton.displayName = 'FakeMoreButton';
+
+const DEFAULT_STATE = {
+  lastVisibleIndex: null,
+  sizes: {tabs: [], more: null}
+};
+
+const ACTIONS = {
+  MEASURE_TABS: 'MEASURE_TABS',
+  DEFINE_LAST_VISIBLE_INDEX: 'DEFINE_LAST_VISIBLE_INDEX'
+};
+
+function visibilityReducer(state, action) {
+  switch (action.type) {
+    case ACTIONS.MEASURE_TABS: {
+      const {more, tabs} = action;
+      return {
+        ...state,
+        sizes: {
+          more,
+          tabs
+        }
+      };
+    }
+    case ACTIONS.DEFINE_LAST_VISIBLE_INDEX: {
+      const {lastVisibleIndex} = action;
+      return {
+        ...state,
+        lastVisibleIndex
+      };
+    }
+    default:
+      throw new Error();
+  }
+}
+
+const useAdjustHandler = ({
+  measureRef, elements, children, selectedIndex, dispatch
+}) => React.useCallback(entry => {
+  const containerWidth = entry ? entry.contentRect.width : measureRef.current.offsetWidth;
```

Comment:
after you removed MutationObserver, `entry` is always truthy, so the `measureRef.current.offsetWidth` part is not needed